.pre-commit-config.yaml

```diff
@@ -9,7 +9,20 @@ repos:
   rev: 22.10.0
   hooks:
   - id: black
+- repo: https://github.com/python-poetry/poetry
+  rev: '1.2.2'
+  hooks:
+    - id: poetry-check
+    - id: poetry-lock
```

Comment:
Is it necessary to run this hook before every commit?

Reply:
Yes, renovate will probably do the main updating. This is so that if manual changes are made it's sure to get synced to the lock file and requirements files.

Reply:
If you want to make sure lock file are synched with pyproject.toml, you only need `poetry-check`

Reply:
poetry-check only checks the structure of pyproject.toml https://python-poetry.org/docs/master/cli/#check

poetry-lock makes sure the lock file is up to date.

It didn't seem too slow while I was committing, a second or so, but I changed it now so that it only runs when pyproject.toml has been changed. Fairly certain that's the only time we'd need to update the lock file, and I'm actually a bit confused why that isn't the default for the hook.

Reply:
I'm embarrassing... 

I'm actually suggesting `poetry-lock` with `--check` args, not currently `--no-update`, in this case it only check if lock file is synced with main file, won't do deps resolving.

Reply:
Hmm. That will fail in the same cases, but not fix the lock file, no? That would speed up the check, but we'd want to run `poery lock` afterward anyway to fix the lock file wouldn't we? I think it's okay that it's a little bit slow, now that I've changed it to only run when pyproject or the lock file changes. It might be a good idea to add `poetry check --lock` as well, so that it runs on the CI as well (where the poetry lock hook is disabled because there is no network), and fails PRs that don't match though...

Reply:
I added the bit for `poetry lock --check` on the CI. What do you think about the current state?

Reply:
It looks fine.

It's just poetry dependencies resolving is really slow, so only check lock in pre-commit hooks and let developer to manually run `poetry lock` after they manually change `pyproject.toml#poetry` is not a bad idea.

This is how I take to run `poetry lock --no-update` on my machine for the first time, without any poetry cache:

```text
$ time poetry lock --no-update

Writing lock file

real    7m56.862s
user    0m0.046s
sys     0m0.015s
```

Reply:
8 minutes! Is that on poetry 1.2.2? My cache must have already been good or something, because I haven't seen that sort of time on any recent poetry versions. I agree though, if it can take that long, we don't want it as part of the hook.

Reply:
I'm using poetry 1.2.1

Reply:
I'll take it out just in case. The failure with --check should be good enough, and we don't want anybody wondering what the hell is taking their commit so long if anything like that can happen.